File: C#/백준/Silver/2217. 로프/로프.cs
using System;
using System.Linq;

namespace q{
    class w{
        static void Main(){
            int a =  int.Parse(Console.ReadLine());
            int[] aa = new int[a];
            for (int i=0;i<a;i++){
                aa[i] = int.Parse(Console.ReadLine());
            }
            Array.Sort(aa);
            for (int i=0;i<a;i++){
                aa[i]*=a-i;
            }
            Console.WriteLine(aa.Max());
        }
    }
}

File: Python/백준/Silver/2217. 로프/로프.py
import sys

a = int(sys.stdin.readline())
aa = []
bb = []
for i in range(a):
    aa.append(int(sys.stdin.readline()))
aa.sort()
for i in range(a):
    bb.append((a-i)*aa[i])
sys.stdout.write(f"{max(bb)}")